CONTRIBUTING.md

```diff
@@ -0,0 +1,147 @@
+
+### `NON_MATCHING`
+
+If you make significant progress on decompiling a function, but can't get it to match perfectly, you can use a `NON_MATCHING` block to commit your code but have it excluded from the build, like so:
+
+```c
+#ifdef NON_MATCHING
+void CollisionCheck_SpawnWaterDroplets(GlobalContext* ctxt, Vec3f* v) {
+    // ... 
+}
+#else
+#pragma GLOBAL_ASM("./asm/non_matchings/code/z_collision_check/CollisionCheck_SpawnWaterDroplets.asm")
+#endif
+```
+
+Before using `NON_MATCHING`, first try to use the [decomp-permuter](tools/decomp-permuter) tool to find a closer match.
+
+### Matching vs. Documenting
+
+Usually, the first step of decompiling a section of code is to get it *matching*: to produce a C version of the code that can be compiled into an identical ROM.
+
+However, the goal of this project is to produce a codebase that can be understood and modified.
+So, beyond producing matching C code, the next steps are *documenting* the code.
+
+Documenting is more than just adding comments, and entails:
+
+- Renaming functions, variables, and struct members
+- Using (or adding) constants, enums, and macros when possible
+- Explaining sections of code that are not straightforward
+
+Overlays are not required to be documented at this time, but `code/` and `boot/` are.
+When an overlay documentation has been started, it will be required to be completed.
+
+For right now, object segment symbols should not be documented/renamed.
+These will be given names when object reconstruction is ready.
+However, it can be useful to add comments with name suggestions.
+
+
+Pull Requests (PRs)
+-------------------
+
+### Checklist
+
+Before opening a PR, walk through the following steps to ensure that your code conforms to the style guide & conventions.
+
+- `./format.sh` was run to apply standard formatting.
+- `make` successfully builds a matching ROM.
+- No new compiler warnings during the build process
+- New variables & functions should follow standard naming conventions.
+- Comments and variables have correct spelling.
+
+### Pull Request Process
+
+After opening a PR, the Jenkins server will verify your changes.
+If there is an error, double-check that you can successfully `make clean && make` locally and that all added/modified files were `git add`-ed to your commit.
+
+Each PR needs a review from one reviewer, plus the project lead.
+
+The PR author marks the comments as resolved: the commenter may not have permissions to do so.
+Once all comments are addressed, it is courteous to ping the reviewer on either Discord or GitHub via re-requesting a review.
+
+After all the comments are addressed and at least one contributor has approved the review, the project lead can then merge the code.
+The project lead is also responsible for ensuring that all of these procedures are followed.
+
+Throughout the process, you (the PR author) should update the [Trello board](https://trello.com/b/ruxw9n6m/majoras-mask-decompilation) with the status of your PR.
```

Comment:
Updating trello includes both putting it in the correct column, and updating it with appropriate labels.